### ISSUE TITLE
Lazily Create Monaco Models in Resource Preference Providers

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-theia/theia#7426]
For a while now, AbstractResourcePreferenceProvider has used monaco editor models to manage the contents of settings files. This creates a circular dependency, because the editor model waits for preferences to be ready and preferences can't be read before the editor model is created. While this seems to work, I think it would be better if we would apply the following changes:

1. Read the file contents directly from the file initially. Only signal "ready" once the file contents have been read
2. Create a monaco model on demand, that is when opening an editor or when setting a preference programatically. Once we have the model, we can keep it just like now.

Opening an editor will create a model, which will wait for preferences to be ready. I also would setting a preference before the preferences provider is ready to be a programming error to be rejected.

**Comment by akosyakov** [comment]
We should check though that's proper documents settings are applied on preference save: https://github.com/eclipse-theia/theia/blob/f5433ece0a32211439fd4acac33200894de1f012/packages/preferences/src/browser/abstract-resource-preference-provider.ts#L131 Editor preferences are used to configure them for monaco models.

**Comment by tsmaeder** [comment]
Yes, we would continue using models for all _changes_ to the preferences, be it programmatically or by an editor. A save of an unchanged file should be a no-op, though (IMO).

**Comment by tsmaeder** [comment]
@colin-grant-work do you think we can close this one now? (I don't remember all the contents of your PR anymore)

**Comment by colin-grant-work** [comment]
@tsmaeder, if you're happy, I'm happy closing it. Technically, the PR didn't do full lazy loading of the Monaco model but just introduced an additional step in the `@postConstruct` to read the preferences using the file system - the model is then still created as part of the `init()` method. It's pretty easy to do full lazy loading on top of that work if you view that as desirable, but to me, getting the load done before `ready` was the biggest issue, and that's taken care of.
